src/core/utils/misc.cpp

```diff
@@ -315,30 +317,57 @@ QString Utils::Misc::pythonVersionComplete() {
     return version;
 }
 
-// return best userfriendly storage unit (B, KiB, MiB, GiB, TiB)
+QString Utils::Misc::unitString(Utils::Misc::SizeUnit unit)
+{
+    return QCoreApplication::translate("misc",
+         units[static_cast<int>(unit)].source, units[static_cast<int>(unit)].comment);
+}
+
+// return best userfriendly storage unit (B, KiB, MiB, GiB, TiB, ...)
 // use Binary prefix standards from IEC 60027-2
 // see http://en.wikipedia.org/wiki/Kilobyte
 // value must be given in bytes
 // to send numbers instead of strings with suffixes
-QString Utils::Misc::friendlyUnit(qreal val, bool is_speed)
+bool Utils::Misc::friendlyUnit(qreal rawVal, qreal &val, Utils::Misc::SizeUnit &unit)
```

Comment:
I guess there are negative sizes somewhere, and they are used to represent unknown size, because someone even created a translatable string for this case.  For the same reason the function has to signal about incorrect size, so it returns ```bool```. 
The very first action of this function will be the conversion of integral input into floating point value, but since everybody want cleaner interface, let's change both function signatures to ```qint64```.